### PR TITLE
Create process table

### PR DESCRIPTION
**Pull request by mckeenicholas**
Creates a basic data structure to keep ownership of PCBs

### CODE REVIEW

File: gibinit
Comment (1whatleytay) [inline gibinit:1]: Probably don't want to commit these two.

**Review by 1whatleytay — APPROVED** [review]

File: kernel/src/threading/scheduling/wait_queue.rs
Comment (ff-zhang) [inline kernel/src/threading/scheduling/wait_queue.rs:1]: Probably showed be renamed, bit of a misnomer to call this a queue

File: programs/wait/.gitignore
Comment (ff-zhang) [inline programs/wait/.gitignore:1]: Should this be included?

File: programs/syscall/syscall
Comment (ff-zhang) [inline programs/syscall/syscall:1]: Binaries shouldn't be included

File: programs/wait/wait.S
Comment (ff-zhang) [inline programs/wait/wait.S:1]: If this is only for testing of wait, I don't think we should include it

**Review by ff-zhang — COMMENTED** [review]
Outside of some extraneous files, it looks good

File: kernel/src/threading/process_table.rs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        assert!(
+            !self.table.contains_key(&pcb.pid),
+            "PCB with pid {} already added to process table.",
+            pcb.pid
+        );
+        self.table.insert(pcb.pid, pcb);
+    }
+
+    pub fn remove(&mut self, pid: Pid) -> Option<Box<ProcessControlBlock>> {
+        #[allow(dead_code)]
```
Comment (ff-zhang) [inline kernel/src/threading/process_table.rs:27]: NIT: Should this be above the function declaration?

**Review by ff-zhang — APPROVED** [review]
LGTM

File: kernel/src/threading/process_table.rs (hunk 1 above)
Comment (mckeenicholas) [reply kernel/src/threading/process_table.rs:27]: I was getting intellisense errors with it above, this is just to make clippy not complain and can be removed once the function is used.

**Pull request merged**